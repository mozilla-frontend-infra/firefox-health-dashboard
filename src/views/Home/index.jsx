import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import DashboardPage from '../../components/DashboardPage';

const styles = {
  root: {
    color: 'white',
    backgroundColor: 'black',
    height: '100%',
  },
  links: {
    lineHeight: '3rem',
    fontSize: '2rem',
    display: 'flex',
    flex: '1',
    flexDirection: 'column',
    padding: '10rem 0 0 0',
    textAlign: 'center',
  },
  netlify: {
    position: 'fixed',
    left: 0,
    bottom: 0,
    width: '100%',
    textAlign: 'center',
  },
};
const Home = ({ classes }) => (
  <DashboardPage
    classes={{
      root: classes.root,
    }}
    title="Firefox health"
    subtitle="Tracking metrics for Firefox products">
    <div className={classes.links}>
      <Link to="/android">Android</Link>
      <Link to="/quantum/32">Quantum 32bit</Link>
      <Link to="/quantum/64">Quantum 64bit</Link>
      <Link to="/js-team">JS team</Link>
    </div>
    <div className={classes.netlify}>
      <a href="https://www.netlify.com">
        <img
          src="https://www.netlify.com/img/global/badges/netlify-dark.svg"
          alt="Netlify"
        />
      </a>
    </div>
  </DashboardPage>
);

Home.propTypes = {
  classes: PropTypes.shape({}).isRequired,
};

export default withStyles(styles)(Home);
