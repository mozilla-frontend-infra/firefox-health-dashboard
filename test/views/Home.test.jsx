import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../../src/views/Home';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
