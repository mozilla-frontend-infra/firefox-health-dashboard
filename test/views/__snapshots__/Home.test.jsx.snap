// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="DashboardPage-root-4 Home-root-1"
>
  <div
    className="DashboardPage-title-5"
  >
    <h1>
      Firefox health
      <small
        className="DashboardPage-subtitle-6"
      >
        Tracking metrics for Firefox products
      </small>
    </h1>
  </div>
  <div
    className="Home-links-2"
  >
    <a
      href="/android"
      onClick={[Function]}
    >
      Android
    </a>
    <a
      href="/quantum/32"
      onClick={[Function]}
    >
      Quantum 32bit
    </a>
    <a
      href="/quantum/64"
      onClick={[Function]}
    >
      Quantum 64bit
    </a>
    <a
      href="/js-team"
      onClick={[Function]}
    >
      JS team
    </a>
  </div>
  <div
    className="Home-netlify-3"
  >
    <a
      href="https://www.netlify.com"
    >
      <img
        alt="Netlify"
        src="https://www.netlify.com/img/global/badges/netlify-dark.svg"
      />
    </a>
  </div>
</div>
`;
